Add platform-dependencies and hashes CLI arguments (#8)
guardicode/agent-plugin-builder: README.md

        -s/--source-dir: The name of the source directory.
        Default: <plugin_name>_<plugin_type>

>         -pd/--platform-dependencies: The platform dependencies for the Agent Plugin.

Reviewer: Can we include the possible values? It isn't documented anywhere but the code for the moment. Is it possible that argparse can display that automagically for us? If not, consider using [click](https://click.palletsprojects.com/en/8.1.x/) instead. I'm fairly confident it generates better documentation for enums.

        -s/--source-dir: The name of the source directory.
        Default: <plugin_name>_<plugin_type>

        -pd/--platform-dependencies: The platform dependencies for the Agent Plugin.
        Options:
        common: All dependencies are packaged once, and shared across all platforms. This
            should only be used if all dependencies are known to be platform-independent.
        separate: Some or all dependencies are platform-dependent, and are therefore packaged
            separately for each supported platform. This is the most reliable option,
            however it results in a larger plugin file, since dependencies are
            duplicated for each platform.
        autodetect: The plugin builder will attempt to detect the best method to use.

>         Default: seperate

Reviewer: Maybe this isn't a question for this PR, but why isn't the default `autodetect`?

Author: @cakekoa ?

        -s/--source-dir: The name of the source directory.
        Default: <plugin_name>_<plugin_type>

        -pd/--platform-dependencies: The platform dependencies for the Agent Plugin.
        Options:
        common: All dependencies are packaged once, and shared across all platforms. This
            should only be used if all dependencies are known to be platform-independent.
        separate: Some or all dependencies are platform-dependent, and are therefore packaged
            separately for each supported platform. This is the most reliable option,
            however it results in a larger plugin file, since dependencies are
            duplicated for each platform.
        autodetect: The plugin builder will attempt to detect the best method to use.
        Default: seperate

>         -h/--hashes/--no-hashes: Specify whether we should verify plugins dependencies.
>         --hashes: Verify the integrity of the plugins dependencies. Recommended
>         --no-hashes: Do not verify the integrity of the plugins dependencies. Strongly
>         not recommended
>         not specified: Same as --hashes.

Reviewer: ```suggestion
        -h/--hashes/--no-hashes: Specify whether to verify the plugin's dependencies.
        --hashes: Verify the integrity of the plugin's dependencies. (Recommended, default)
        --no-hashes: Do not verify the integrity of the plugin's dependencies. (Not recommended)
        not specified: Same as --hashes.
```

Reviewer: Should we name the option `--verify/--no-verify`?
